### DISCUSSION TITLE
Got error from .env...Error: Failed to decode private key

### DISCUSSION THREAD

**Discussion by DjiaaX** [Cyfrin/foundry-full-course-cu#2799] [Q&A] [answered]
I dunno how to solve this, when i type this in my terminal "forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --private-key $PRIVATE_KEY --broadcast", this happen..Error: Failed to decode private key. Feel like im already follow the course correctly, and i try to ask GPT but ITS still the same🥲
I have to export the private key first before i run that command..like "export $PRIVATE_KEY=blablabla" after that i can compile correctly, hope someone help me about this error.
(Im sorry if the text is too long XD)

**Comment by LordAdhaar** [comment]
Try using 
`source .env`
to export the variable

Use 
`echo $PRIVATE_KEY`
to check the value

Instead of storing your private keys in .env a better way is 
`forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --interactives 1`
and enter your private key

**Comment by EngrPips** [comment, marked as answer]
Hello @DjiaaX, Please read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2299. I think it will help you

**Reply by DjiaaX** [reply]
Wow!! Its really works.. thankyou SM (⁠≧⁠▽⁠≦⁠)

**Reply by EngrPips** [reply]
Awesome, I am glad that worked. Keep crushing it.
